csrc/ir/nodes.cpp

```diff
@@ -4380,6 +4391,12 @@ std::vector<PolymorphicValue> SdpaFwdOp::evaluate(
               /*return_debug_mask=*/false,
               scale);
 
+  // Add back the device dim axis for outputs with a head dimension.
+  if (handle_device_dim) {
+    output = output.unsqueeze(0);
+    log_sumexp = log_sumexp.unsqueeze(0);
```

Comment:
Why do we need the DID axis in `log_sumexp`?

Reply:
Good question, we only need it for the backwards, so we don't need to track the DID axis. I'll remove it!